crates/ty_python_semantic/resources/mdtest/del.md

```diff
@@ -239,21 +239,43 @@ del g[0]
 ### TypedDict deletion
 
 Deleting a required key from a TypedDict is a type error because it would make the object no longer
-a valid instance of that TypedDict type.
+a valid instance of that TypedDict type. However, deleting `NotRequired` keys (or keys in
+`total=False` TypedDicts) is allowed.
 
 ```py
-from typing_extensions import TypedDict
+from typing_extensions import TypedDict, NotRequired
 
 class Movie(TypedDict):
     name: str
     year: int
 
+class PartialMovie(TypedDict, total=False):
+    name: str
+    year: int
+
+class MixedMovie(TypedDict):
+    name: str
+    year: NotRequired[int]
+
 m: Movie = {"name": "Blade Runner", "year": 1982}
+p: PartialMovie = {"name": "Test"}
+mixed: MixedMovie = {"name": "Test"}
 
+# Required keys cannot be deleted.
 # error: [invalid-argument-type]
```

Comment:
I think it's cool that we do this by synthesizing overloads, but what does this error end up looking like? Can we snapshot-diagnostics here?